src/TestData/Factories/DonationFactory.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace GiveTestData\TestData\Factories;
+
+use GiveTestData\TestData\Framework\Factory;
+
+/**
+ * Class Donation
+ * @package GiveTestData\TestData\Factories
+ */
+class DonationFactory extends Factory {
+	/**
+	 * @var string
+	 */
+	private $status;
+	/**
+	 * @var int
+	 */
+	private $amount;
+
+	/**
+	 * @var string
+	 */
+	private $currency;
+
+	/**
+	 * @param string $status
+	 */
+	public function setDonationStatus( $status ) {
+		$this->status = $status;
+	}
+
+	/**
+	 * Check is valid donation status
+	 *
+	 * @param string $status
+	 *
+	 * @return bool
+	 */
+	public function checkDonationStatus( $status ) {
+		if ( 'random' === $status ) {
+			return true;
+		}
+
+		return in_array( $status, array_keys( give_get_payment_statuses() ) );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getDonationStatus() {
+		if ( is_null( $this->status ) ) {
+			return 'publish';
+		}
+
+		if ( 'random' === $this->status ) {
+			return $this->randomDonationStatus();
+		}
+
+		return $this->status;
+	}
+
+	/**
+	 * @param int $amount
+	 */
+	public function setDonationAmount( $amount ) {
```

Comment:
@kjohnson they are used here https://github.com/impress-org/givewp-test-data/blob/0ce77a905b514e8243f9a243e2d7a498b62c6f24/src/TestData/Commands/DonationSeedCommand.php#L93

Why I went with this route in the first place? I thought that it would be beneficial to have more options when seeding the data using CLI or UI.  For example, in E2E testing, it's good to have predictable results, like the Revenue amount. These setters allow me to easily set various donation options.